docs/book/v1/chapter-3.md

```diff
@@ -0,0 +1,454 @@
+}
+
+/**
+@return array{
+ *     paths: array{page: array{literal-string&non-falsy-string}}
+ * }
+ */
+private function getTemplates(): array
+{
+    return [
+        'paths' => [
+            'page' => [__DIR__ . '/../templates/page'],
+        ],
+    ];
+}
+```
+
+In Dotkernel (and Mezzio/Laminas in general), most classes cannot be created directly with new ClassName() because they require dependencies (such as the template engine or a repository).
+To provide those dependencies, we create Factory classes.
+
+A factory’s only responsibility is to build an object with everything it needs.
+For that, we have created two factories - one for our Handler and one for our Repository.
+
+`src/App/src/Factory/BookHandlerFactory.php`:
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace Light\App\Factory;
+
+use Light\Book\Handler\GetBooksHandler;
+use Light\Book\Repository\BookRepository;
+use Mezzio\Template\TemplateRendererInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+use function assert;
+
+class BookHandlerFactory
+{
+    /**
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function __invoke(ContainerInterface $container, string $requestedName): GetBooksHandler
+    {
+        $repository = $container->get(BookRepository::class);
+        $template   = $container->get(TemplateRendererInterface::class);
+
+        assert($repository instanceof BookRepository);
+        assert($template instanceof TemplateRendererInterface);
+
+        return new GetBooksHandler($template, $repository);
+    }
+}
+```
+
+`src/App/src/Factory/BookRepositoryFactory.php`:
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace Light\App\Factory;
+
+use Doctrine\ORM\EntityManager;
+use Light\Book\Entity\Book;
+use Light\Book\Repository\BookRepository;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+use function assert;
+
+class BookRepositoryFactory
+{
+    /**
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function __invoke(ContainerInterface $container): BookRepository
+    {
+        $entityManager = $container->get(EntityManager::class);
+
+        $repository = $entityManager->getRepository(Book::class);
+        assert($repository instanceof BookRepository);
+
+        return $repository;
+    }
+}
+```
+
+All the changes are now registered.
+
+## Populating the Table
+
+In order to have values for us to display, we now have to populate our `books` table from our database with some values.
+There are two options in this case:
+
+- Use Doctrine Fixtures
+- Add entries manually from the Database Client.
+
+**We recommend using fixtures**, but if you choose the alternative approach, feel free to skip directly to the **final section** of this chapter.
```

Comment:
Instead of using **final section**, you should use an anchor that takes the reader to that section.
[anchor example](https://github.com/dotkernel/development/blob/alma-linux-10/docs/book/v2/setup/installation.md?plain=1#L36)